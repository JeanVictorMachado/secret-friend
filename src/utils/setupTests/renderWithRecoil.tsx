import { render } from "@testing-library/react";
import { ReactElement } from "react";
import { RecoilRoot } from "recoil";

const renderWithRecoil = (ui: ReactElement) => {
  const wrapper = <RecoilRoot>{ui}</RecoilRoot>;

  return render(wrapper);
};

export * from "@testing-library/react";
export { renderWithRecoil };
