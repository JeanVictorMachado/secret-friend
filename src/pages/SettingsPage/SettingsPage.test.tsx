import { renderWithRecoil } from "../../utils/setupTests/renderWithRecoil";

import { SettingsPage } from ".";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigate,
  };
});

describe("Settings Page", () => {
  it("The page must be render correctly", () => {
    const { container } = renderWithRecoil(<SettingsPage />);

    expect(container).toMatchSnapshot();
  });
});
