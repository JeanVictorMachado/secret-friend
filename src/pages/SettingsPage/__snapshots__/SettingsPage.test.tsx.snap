// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page The page must be render correctly 1`] = `
<div>
  <div
    class="sc-bcXHqe fCbXqd"
  >
    <section>
      <h2>
        Vamos come√ßar!
      </h2>
      <form>
        <div
          class="sc-hLBbgP fhqSRt"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul
        class="sc-jSUZER"
      />
      <footer
        class="sc-gswNZR kBhDsb"
      >
        <button
          class="sc-dkrFOg gxloLz"
          disabled=""
        >
          Iniciar brincadeira
        </button>
        <img
          alt="Sacolas de compras"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
